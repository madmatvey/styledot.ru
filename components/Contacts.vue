<template>
  <div class="columns">
    <div
      v-for="(messenger, index) in contactsOptions"
      :key="index"
      class="column"
      @click="onClickLink(rep, index)"
    >
      <a :href="messenger.urlLink" @click="onClickLink(messenger.name)">
        <figure class="image is-48x48 mb-2 contactsIcon">
          <img :src="'/images/' + messenger.name.toLowerCase() + '.svg'" />
        </figure>
      </a>
      <div class="block is-size-6 has-text-centered">{{ messenger.name }}</div>
    </div>
  </div>
</template>

<script>
export default {
  data() {
    return {
      contactsOptions: [
        { name: 'Telegram', urlLink: 'https://t.me/styledot_leonteva' },
        { name: 'WhatsApp', urlLink: 'https://wa.me/79277461266' },
        { name: 'Viber', urlLink: 'viber://chat?number=79277461266' },
      ],
    }
  },
  methods: {
    onClickLink(name) {
      this.$ga.event({
        eventCategory: 'Contact Me Link',
        eventAction: 'click',
        eventLabel: name,
        eventValue: 0,
      })
    },
  },
}
</script>

<style>
.contactsIcon {
  justify-content: center;
  margin: auto;
}
</style>
