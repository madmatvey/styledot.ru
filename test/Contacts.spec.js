import { shallowMount, mount } from '@vue/test-utils'
import Contacts from '@/components/Contacts.vue'

const factory = () => {
  return shallowMount(Contacts, {})
}

describe('Contacts', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(Contacts)
    expect(wrapper.vm).toBeTruthy()
  })

  test('renders properly', () => {
    const wrapper = factory()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
