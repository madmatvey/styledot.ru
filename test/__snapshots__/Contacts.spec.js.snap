// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts renders properly 1`] = `
"<div class=\\"columns is-mobile\\">
  <div class=\\"column is-one-third-mobile\\"><a href=\\"https://t.me/styledot_leonteva\\">
      <figure class=\\"image is-48x48 mb-2 contactsIcon\\"><img src=\\"/images/telegram.svg\\"></figure>
    </a>
    <div class=\\"block is-size-7 has-text-centered\\">Telegram</div>
  </div>
  <div class=\\"column is-one-third-mobile\\"><a href=\\"https://wa.me/79277461266\\">
      <figure class=\\"image is-48x48 mb-2 contactsIcon\\"><img src=\\"/images/whatsapp.svg\\"></figure>
    </a>
    <div class=\\"block is-size-7 has-text-centered\\">WhatsApp</div>
  </div>
  <div class=\\"column is-one-third-mobile\\"><a href=\\"viber://chat?number=79277461266\\">
      <figure class=\\"image is-48x48 mb-2 contactsIcon\\"><img src=\\"/images/viber.svg\\"></figure>
    </a>
    <div class=\\"block is-size-7 has-text-centered\\">Viber</div>
  </div>
</div>"
`;
